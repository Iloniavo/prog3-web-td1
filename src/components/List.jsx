
export default function List({children, title, listTestId, headTestId}) {
    return (
        <div  data-testid="list-content">
            <h2 data-testid={headTestId} >{title}</h2>
            <ul data-testid={listTestId}>
                {children}
            </ul>
        </div>
    )

};
 