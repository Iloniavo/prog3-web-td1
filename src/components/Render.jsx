
import React from 'react';
import { useState } from 'react';
import Input from './Input';
import List from './List';
import ListItem from './ListItem';

export default function Render() {
    const [ list, setList ] = useState([]);    
    const [ description, setDescription ] = useState("")
    const [ id, setId ] = useState(0);


    function checkInput(e){
        e.preventDefault()
        description === "" ? alert("Check your input") : addItem(e)
    }

    function addItem(e){
        e.preventDefault()
       
        let item = {
            id: id,
            description: description,
            done: false
        }
        console.table("all in one "+item);
        setList([...list, item])
        setDescription("");
        setId(id + 1)
    
    }

    return (
        <div>

            <form action="" onSubmit={(e) => checkInput(e)} >
                <Input description={description} setDescription={setDescription} />
                <button type='submit' data-testid="create-button" >Ajouter</button>
            
            </form>
            <List title={"To do :"}  listTestId= "todo-content" headTestId={"todo-header"}>
                    {list.filter((item) => item.done === false ).map((item, i) => 
                        <ListItem key={ `${item.description} - ${item.id}` } list={list} setList={setList} i={item.id} description={item.description}/>
                    )}
            </List> 
            <List title={ "Done :" } listTestId="done-content" headTestId={"done-header"} >
                 {list.filter(item => item.done === true).map((item, index) => <li key={index} >{item.description}</li>)}
            </List>           
        
        </div>
    )

};
