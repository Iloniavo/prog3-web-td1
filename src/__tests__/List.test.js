import renderer from "react-test-renderer";
import {fireEvent, render, screen} from "@testing-library/react";
import List from "../components/List";
import Render from "../components/Render";

describe("Test list component", function (){

    it("Empty list", function (){
        const component = renderer.create(
            <List title={"To do"} headTestId={"todo-header"}/>
        )
             
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
        render( <List title={"To do"} headTestId={"todo-header"} /> )
     
        expect(screen.getByTestId("todo-header")).toHaveTextContent("To do")    
    }) 
     
    it("Render to do list", function (){
        const { getByTestId } = render(
            <List title={"To do"} listTestId={"todo-content"}>
                <li>desc</li>
            </List>
        )
        expect(getByTestId("todo-content")).toHaveTextContent("desc")

    })
    
    it("Render done list", function(){
        const { getByTestId } = render(
            <List title={"Done"} listTestId={"done-content"}>
                <li>desc</li>
            </List>
        )
        expect(getByTestId("done-content")).toHaveTextContent("desc")
    })

    it("add content to to do list", function (){
        const component = render(<Render/>)

        const inputElement = component.getByTestId("input")
        const createButton = component.getByTestId("create-button")

        fireEvent.change(inputElement, { target : { value : "desc" } })
        fireEvent.submit(createButton);
 
        const todo = component.getAllByTestId("todo-content")
        const todoFirstElement = todo[0];

        expect(todoFirstElement.textContent).toBe("desc")

        expect(todo[0]).toBeInTheDocument();

    })

}) 

