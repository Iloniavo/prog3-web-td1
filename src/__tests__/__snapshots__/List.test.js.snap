// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test list component Empty list 1`] = `
<div
  data-testid="list-content"
>
  <h2
    data-testid="todo-header"
  >
    To do
  </h2>
  <ul />
</div>
`;

exports[`Test list component add content to done list 1`] = `
<div
  data-testid="list-content"
>
  <h2 />
  <ul>
    <li>
      desc
    </li>
  </ul>
</div>
`;

exports[`Test list component add content to done list 2`] = `"desc"`;
